sdk/php/src/Command/CodegenCommand.php

```diff
@@ -30,12 +32,31 @@ public function __construct()
         $this->daggerConnection = Connection::get();
     }
 
+    public function configure(): void
+    {
+        $this->addOption(
+            'schema-file',
+            null,
+            InputArgument::OPTIONAL,
+            'Path to the schema json file',
+            null
+        );
+    }
+
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
         $io = new SymfonyStyle($input, $output);
-        $client = $this->daggerConnection->connect();
 
-        $schema = (new SchemaGenerator($client))->getSchema();
+
+        if ($input->getOption('schema-file') !== null && file_exists($input->getOption('schema-file'))) {
+            $fileContents = file_get_contents($input->getOption('schema-file'));
+            $schemaArray = json_decode($fileContents, true);
```

Comment:
Change this to
```
json_validate()
https://www.php.net/json_validate
We need PHP 8.3 on this, now
```

Reply:
From [php.net/json_validate](https://www.php.net/json_validate)

> Caution
> 
> Calling json_validate() immediately before [json_decode()](https://www.php.net/manual/en/function.json-decode.php) will unnecessarily parse the string twice, as [json_decode()](https://www.php.net/manual/en/function.json-decode.php) implicitly performs validation during decoding.
> 
> json_validate() should therefore only be used if the decode JSON payload is not immediately used and knowing whether the string contains valid JSON is needed.

https://github.com/dagger/dagger/blob/c5bcedfab3a0f6bd32fd4bb80b8e6b5612037a8a/sdk/php/src/Command/CodegenCommand.php#L53-L54

I might be misunderstanding the intent, but the decoded JSON payload is immediately used. Is it still worth using `json_validate` in this instance?